### PR TITLE
Feat/Expose user subscription streak

### LINKED ISSUE #15: User should be able to see their streak as subscriber
**What:** User should be able to see their streak as a subscriber

**Why:** We want to add gamification to subscriptions by adding a streak to a user subscription. The idea is to awards badges and other things to users in our community first and then integrate it with the rest of our apps.
 
**How:**

- We can get the streak (months with an active subscription) for each user.
- Other applications can query for this parameter easily through API

### LINKED ISSUE #18: User should be able to see their donation history
**What:** User should be able to see their donation history

**Why:** For users to know what have been their contributions to the Debt Collective. 

**How:**

- Implement a section in the back-office where users can see all the donations to date.

### PR DESCRIPTION
**Pull request by castrolem**
**What:**
Add helper method in `User` to get the current streak through the `Subscription.start_date` DB field.

The public endpoint to access a user streak URL is `GET    /users/:user_id/streak(.:format)`

**Why:**
To show the subscribed user, how much time he/she has been subscribed.
Closes #15 and  closes #18 

**How:**
Through an API endpoint for public consumption and a view in the user profile.

**To Do:**
* [x] Add user profile view.
* [x] Add styles using React and Material UI for all User views.

## Screenshots
image

### CODE REVIEW

**Comment by orlando** [comment]
@castrolem can you add some screenshots to the PR when you can please? Thanks!

File: app/controllers/streaks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class StreaksController < ApplicationController
+  before_action :set_user, only: %i[show]
+
+  def show
+    render json: {current_streak: @user.current_streak}
```
Comment (orlando) [inline app/controllers/streaks_controller.rb:7]: @castrolem should we add this to the subscription instead? I don't see a case where we need to query just the streak

Comment (castrolem) [reply app/controllers/streaks_controller.rb:7]: I thought it could be a bit of too much data to be available. What's your opinion?

**Comment by castrolem** [comment]
@orlando fixed all conflicts.

**Comment by castrolem** [comment]
@orlando I will merge this PR cuz it's blocking other work, let me know later the comments.

**Pull request merged**